### PR TITLE
feat: export transform API

### PR DESCRIPTION
**Pull request by dherges**
## I'm submitting a...

```
[ ] Bug Fix
[x] Feature
[ ] Other (Refactoring, Added tests, Documentation, ...)
```

## Checklist

- [x] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [ ] Tests for the changes have been added

## Description

WIP see #643 

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/lib/ng-v5/entry-point/resources/stylesheet.transform.ts [hunk 1]
```diff
@@ -181,3 +182,37 @@ const renderStylus = ({ filename, root, paths }): Promise<string> => {
         })
     );
 };
+
+export interface ForEachStylesheetFn {
+  (css: string, filePath: string, entryPoint: NgEntryPoint): string | Promise<string>;
+}
+
+export function forEachStylesheet(fn: ForEachStylesheetFn): Transform {
```
Comment (dherges) [inline src/lib/ng-v5/entry-point/resources/stylesheet.transform.ts]: We need a test here that verifies the correct behaviour.

- [ ] create a mock graph and feed some nodes to the transform
- [ ] verify that `fn` is called for every stylesheet
- [ ] verify that result from `fn` is applied to stylesheet nodes

File: src/lib/ng-v5/packagr.spec.ts [hunk 2]
```diff
@@ -48,6 +52,41 @@ describe(`ngPackagr()`, () => {
...
+
+          done();
+        });
+    });
+
+    it(`should return self instance for chaining`, () => {
+      expect(packager.withTsConfigTransform(mockTransform)).to.equal(packager);
+    });
+  });
 });
```
Comment (dherges) [inline src/lib/ng-v5/packagr.spec.ts]: We need additional tests for

- [ ] `withStylesheetTransform()`
- [ ] `withTemplateTransform()`

**Comment by fauverism** [comment]
I'm wondering if this is still active or another solution is be favored instead. Does anyone know?

**Comment by gquinteros93** [comment]
Hi @dherges , this solution is what I need to customize my Angular Library css and HTML files during compilation.

Thanks for your work.

Is it possible to merge it and release it?

Thanks in advance.

**Pull request closed without merging**